src/fmu/dataio/_utils.py

```diff
@@ -432,3 +432,37 @@ def read_metadata_from_file(filename: str | Path) -> dict:
         raise OSError(f"Cannot find requested metafile: {metafile}")
     with open(metafilepath) as stream:
         return yaml.safe_load(stream)
+
+
+def get_geometry_ref(
+    geometrypath: str | None, obj: Any
+) -> tuple[str | None, str | None]:
+    """Get a reference to a geometry.
+
+    Read the metadata file for an already exported file, and returns info like this
+    for the data block:
+
+    data:
+      geometry:
+        name: somename
+        relative_path: some_relative/path/geometry.roff
+
+    This means that the geometry may be 'located' both on disk (relative path) and in
+    Sumo
+    """
+    if not geometrypath:
+        return None, None
+
+    gmeta = read_metadata_from_file(geometrypath)
+
+    # some basic checks (may be exteneded to e.g. match on NCOL, NROW, ...?)
+    if isinstance(obj, xtgeo.GridProperty) and gmeta["class"] != "cpgrid":
+        raise ValueError("The geometry for a grid property must be a grid")
+
+    if isinstance(obj, xtgeo.RegularSurface) and gmeta["class"] != "surface":
+        raise ValueError("The geometry for a surface must be another surface")
+
+    geom_name = gmeta["data"].get("name")
+    relpath = gmeta["file"]["relative_path"]
+
+    return geom_name, relpath
```

Comment:
It could be possible to return a pydantic Geometry instance here, but want to avoid that the `_utils` module becomes to much dependant on other modules (i.e. circular dependencies)